### ISSUE TITLE
No field for entering Extra Meals on the Meal Form. 

### ISSUE THREAD

**Issue by crw** [joyvuu-dave/comeals#8]

**Comment by joyvuu-dave** [comment]
I was thinking this is what the max field would be for. Perhaps it'd be nice to generate an Extra Meals value dynamically with max - num. of attendees?

**Comment by crw** [comment]
Let's think about the process flow:

1. Alice and Bob sign up to cook four weeks in advance of the meal. 
1. Alice posts a menu 72 hours before the meal.
1. Residents (let's say 17) sign up. 
1. Alice closes signup 48 hours before the meal. Alice notes the number of guests (17).
1. Bob goes shopping. The recipe rounds well to 20, so he buys enough for 20. 

This is where we split from current process. Today, with the paper form:

1. Bob does the math and writes "3" in the extras spot.

You are proposing:

1. Bob does not do the math and writes "20" in the Max Attendees slot. 

I prefer your solution, but with the following caveats:

1. Today, you seemingly cannot add attendees if the Max is not set. This is a bug, since the Max is not calculated and set until after Close and after Shopping. In fact, Max should not be settable until the meal is Closed. It could be a required value for Closing a meal. (Pre-filled with the # of attendees, when in doubt choose the default and then update after Shopping.) <-- do we need a new issue for this? :)
1. "Extras" or "Remaining Attendee Slots" or whatever we want to call it should be calculated in-line so as to be as not-confusing as possible, per your comment. 

Make sense?

**Comment by joyvuu-dave** [comment]
Thanks for a detailed analysis, I totally agree! It doesn't make sense to set a max until the meal is locked. And that max should automatically be the number of people signed up. 

Why not just make it exactly like the current sheet? We could have meals lock automatically 48hrs before  the meal, and at that point - but not before - the meal form would display an editable extras field that would be set to 0 by default.

**Comment by joyvuu-dave** [comment]
Also, I've fixed the bug so users can now sign up if the max isn't set.

**Comment by crw** [comment]
:+1: 

I am good with your proposal. 

With re: automated Close time, I am not convinced on that. Should we open another issue to discuss? Basically, 48 hours may not be a "one size fits all" type of thing, some may not go shopping until 8 hours before and are happy to leave it open until then. I don't really know. ¯\_(ツ)_/¯ )

**Comment by joyvuu-dave** [comment]
~~You're right; there's no point in having meals close automatically.~~ I just remembered Ann saying she _wanted_ to have her meals closed automatically. What if we added an auto-close checkbox to the meal form so it was an opt-in thing?

**Comment by crw** [comment]
Yeah that works for me. It could even be opt-out, as I think most people would be happy with auto-close.
